src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5199,6 +5199,36 @@ private BoundExpression BindCollectionExpression(CollectionExpressionSyntax synt
 
             MessageID.IDS_FeatureCollectionExpressions.CheckFeatureAvailability(diagnostics, syntax, syntax.OpenBracketToken.GetLocation());
 
+            for (int i = 1, n = syntax.Elements.Count - 1; i < n; i++)
+            {
+                var element = syntax.Elements[i];
+                if (element is WithElementSyntax withElement)
+                {
+                    Error(diagnostics, ErrorCode.ERR_WithElementMustBeFirst, withElement.WithKeyword.GetLocation());
+                    break;
+                }
+            }
+
+            foreach (var element in syntax.Elements)
+            {
+                if (element is KeyValuePairElementSyntax keyValuePairElement)
+                {
+                    MessageID.IDS_FeatureDictionaryExpressions.CheckFeatureAvailability(diagnostics, syntax, keyValuePairElement.ColonToken.GetLocation());
+
+                    // PROTOTYPE: Error for now.  Flesh this out when we do the binding for kvp elements.
+                    Error(diagnostics, ErrorCode.ERR_SyntaxError, keyValuePairElement.ColonToken, ",");
+                    return new BoundBadExpression(syntax, LookupResultKind.Empty, symbols: [], childBoundNodes: [], CreateErrorType());
+                }
+                else if (element is WithElementSyntax withElement)
+                {
+                    MessageID.IDS_FeatureCollectionExpressionConstructionArguments.CheckFeatureAvailability(diagnostics, syntax, withElement.WithKeyword.GetLocation());
```

Comment:
>IDS_FeatureCollectionExpressionConstructionArguments [](http://example.com/codeflow?start=30&length=52)

Are we testing this?

Reply:
likely not. t his was just the parsing PR. binding/testing was intended for you to do :)